tests/host_tools/cargo_build.py

```diff
@@ -33,9 +33,9 @@ def cargo_build(path, extra_args="", src_dir="", extra_env=""):
     """Trigger build depending on flags provided."""
     cmd = "CARGO_TARGET_DIR={} {} cargo build {}".format(path, extra_env, extra_args)
     if src_dir:
-        cmd = "cd {} && {}".format(src_dir, cmd)
-
-    utils.run_cmd(cmd)
+        utils.run_cmd(cmd, cwd=src_dir)
+    else:
+        utils.run_cmd(cmd)
```

Comment:
I think if we defaulted `src_dir =None` in the function signature, this else would not be needed

Reply:
Yup, but this code is yeeted in a follow up commit anyway :)